src/UI/Panels/ClipboardPanel.cs

```diff
@@ -93,21 +151,60 @@ protected override void ConstructPanelContent()
             // Clear clipboard button
             UniverseLib.UI.Models.ButtonRef clearButton = UIFactory.CreateButton(firstRow, "ClearPasteButton", "Clear Clipboard");
             UIFactory.SetLayoutElement(clearButton.Component.gameObject, minWidth: 120, minHeight: 25, flexibleWidth: 0);
-            clearButton.OnClick += () => Copy(null);
-
-            // Current Paste info row
-            GameObject currentPasteHolder = UIFactory.CreateHorizontalGroup(ContentRoot, "SecondRow", false, false, true, true, 0,
-                new(2, 2, 2, 2), childAlignment: TextAnchor.UpperCenter);
-
-            // Actual current paste info label
-            CurrentPasteLabel = UIFactory.CreateLabel(currentPasteHolder, "CurrentPasteInfo", "not set", TextAnchor.UpperLeft);
-            UIFactory.SetLayoutElement(CurrentPasteLabel.gameObject, minHeight: 25, minWidth: 100, flexibleWidth: 999, flexibleHeight: 999);
-            UpdateCurrentPasteInfo();
-
-            // Inspect button
-            UniverseLib.UI.Models.ButtonRef inspectButton = UIFactory.CreateButton(currentPasteHolder, "InspectButton", "Inspect");
+            clearButton.OnClick += ClearClipboard;
+            
+            // Inspect Selected button
+            UniverseLib.UI.Models.ButtonRef inspectButton = UIFactory.CreateButton(firstRow, "InspectButton", "Inspect");
             UIFactory.SetLayoutElement(inspectButton.Component.gameObject, minHeight: 25, flexibleHeight: 0, minWidth: 80, flexibleWidth: 0);
             inspectButton.OnClick += InspectClipboard;
+            
+            // Add New button
+            UniverseLib.UI.Models.ButtonRef addNewButton = UIFactory.CreateButton(firstRow, "AddNewButton", "Add");
+            UIFactory.SetLayoutElement(addNewButton.Component.gameObject, minHeight: 25, flexibleHeight: 0, minWidth: 80, flexibleWidth: 0);
+            addNewButton.OnClick += AddNewEntry;
+
+            RefreshList();
+        }
+
+        public void RefreshList()
+        {
+            foreach (GameObject go in clipboardRows)
+            {
+                //Remove the row
+                GameObject.Destroy(go);
```

Comment:
I don't think it's a good idea to recreate it every time, so can't you use `CreateVerticalGroup` to add elements, etc.?

Reply:
I don't think redrawing it is the most efficient, but I'm not sure how we could do checks to make sure the list UI is always up to date with the actual list. For example, if the GameObject at index 0 is destroyed, how do we make sure that gets updated in the UI? I thought it might be more efficient to just redraw the UI instead of having a List of UI objects and doing stuff with those.

If you don't think it would be more efficient, just lemme know and we can probably figure out a better solution.

Reply:
I feel the implementation of `InspectorManager.CreateInspector` would be helpful
It seems to me that by following that implementation, the Singleton class would not need to be implemented (it could be avoided by making the list part(`CreateVerticalGroup` gameobject) that holds the items static).